### ISSUE TITLE
Lancing of boolean default value in function header

### ISSUE THREAD

**Issue by leloup314** [LeviBorodenko/lancer#7]
When trying to lance `lancer` itself I stumbled upon following output:

`def snarl_Il11l1 (chirp_chirp_chirp_aααaa :Path ="./file.py",roar_αaaαa :bool =hoot_IIΙΙIΙ ):`

which was generated from line [`69`](https://github.com/LeviBorodenko/lancer/blob/master/src/lancer/entry.py#L69) (_nice_) in `src/lancer/entry.py`:

`def lance(file : Path = "./file.py", yolo : bool = False):`

Here, the default value of the argument `yolo=False` was `lanced` which breaks the code.

**Comment by leloup314** [comment]
Minimal example:

    def test_func(val1: str, val2: bool = True):
        if val2:
            print(val1)
  
    if __name__ == '__main__':
        test_func("Uno, Dos, Très, dale", val2=True)

This fails after lancing due to `val2` default value `True` being lanced. When not giving a default value it works after lancing:

    def test_func(val1: str, val2: bool):
            if val2:
                print(val1)
      
    if __name__ == '__main__':
        test_func("Uno, Dos, Très, dale", val2=True)

**Comment by LeviBorodenko** [comment]
Lancing lancer! What an idea! This could be used for testing and maybe we should just have a `lanced-master` branch for giggles.
Frankly, The way variable substitution is implemented is very much suboptimal. It relies on `tokenize` which is very limited. Back in the day I was not aware of the much better suited `ast` module.

While one could patch this issue. I believe that a fundamental rebuild is necessary to truly fix such issues in the long run.

**Comment by leloup314** [comment]
I also thought to myself, actually having the `master` lanced and building wheels for pipy from it would be hilarious and at the same time a nice selling point! One could have a persistant `dev` branch to which PRs are made and on which actual development happens. Then use `github-actions` to automatically push a lanced version of `dev` to `master` whenever a PR is merged / commit is made / version number increased and also build wheels from the lanced version and upload to pipy.

**Comment by LeviBorodenko** [comment]
Absolutely. Actually having lancing in the tool's CI/CD pipeline would be hilarious. But I fear that in the current state the bugs in variable substitution would mess things up.

**Comment by leloup314** [comment]
Maybe this is also useful for people stumbling upon #5
